C++/include/algorithm/dynamic_programming/cut_the_rod.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+    Cutting the rod problem
+    --------------------
+	Given a rod of length n inches and an array of prices 
+	that contains prices of all pieces of size smaller than n. 
+	Determine the maximum value obtainable by cutting up 
+	the rod and selling the pieces
+
+    Time complexity
+    ---------------
+    O(N^2)
+
+    Space complexity
+    ----------------
+    O(N)
+
+    Author
+    ------
+    Fedurko Nikolaus (@B1Z0N)
+*/
+
+#include <vector>	// vector
+
+#ifndef CUT_THE_ROD_HPP
+#define CUT_THE_ROD_HPP
+
+/**
+ * Recursive helper function
+ */
+int __cut_the_rod( const std::vector<int>& costs, std::vector<int>& visited, int len)
```

Comment:
The parameter `len` can be renamed to `length`, and its type (and of the loop variable `i`) should be changed to `std::size_t`.